### PR TITLE
Fix printers test

### PR DESCRIPTION
**Pull request by SkySkimmer**
Maybe next time we will notice it breaking...

### CODE REVIEW

**Comment by ejgallego** [comment]
I did notice it breaking however I didn't have time to investigate, isn't there a better to check that include did work?

**Comment by SkySkimmer** [comment]
The change I made is my best guess to test that it works.

**Comment by ejgallego** [comment]
Umm, see https://discuss.ocaml.org/t/testing-use-directives/3957

In fact in OCaml `trunk` it seems that the toplevel is more strict with `#use` errors.

**Comment by SkySkimmer** [comment]
>In fact in OCaml trunk it seems that the toplevel is more strict with #use errors.

No, look at the new error.

**Comment by ejgallego** [comment]
> No, look at the new error.

At least on master we get:
```
File "/builds/coq/coq/dev/base_include", line 145, characters 5-14:
145 | open Discharge
           ^^^^^^^^^
Error: Unbound module Discharge
# 
0m0.000000s 0m0.000000s
0m0.740000s 0m0.100000s
==========> FAILURE <==========
    misc/printers.sh...Error!
```
and the test fails, so I assumed the behavior had changed, but as I said I didn't investigate.
https://gitlab.com/coq/coq/-/jobs/234756796/artifacts/browse/test-suite/logs/misc/

**Comment by SkySkimmer** [comment]
>At least on master we get:

on edge:trunk:make, but the other jobs get `cannot find include` which wasn't detected by the script.

File: toplevel/coqinit.ml [hunk 1]
```diff
@@ -118,7 +118,7 @@ let init_ocaml_path () =
   let open Loadpath in
   let lp s = { recursive = false; path_spec = MlPath s } in
   let add_subdir dl =
-    Loadpath.add_coq_path (lp (List.fold_left (/) Envars.coqroot [dl]))
+    Loadpath.add_coq_path (lp (List.fold_left (/) (Envars.coqlib()) [dl]))
```
Comment (ejgallego) [inline toplevel/coqinit.ml]: This deserves its own commit and a better explanation, please also use more descriptive commit messages.

**Comment by SkySkimmer** [comment]
I guess when built with dune we should do `#use "include_dune";;`, is there a good way of detecting which case we're in?

**Comment by ejgallego** [comment]
> I guess when built with dune we should do `#use "include_dune";;`, is there a good way of detecting which case we're in?

`INSIDE_DUNE` envar is set for example.

**Comment by SkySkimmer** [comment]
Can you figure out how to get the test to work in dune? I'm lost.

**Comment by ejgallego** [comment]
> Can you figure out how to get the test to work in dune? I'm lost.

I guess the dependencies need to be added to `test-suite/dune` `deps`?

Keep in mind that the test-suite is run in the sandboxed environment.

**Comment by ejgallego** [comment]
Ok attempt to fix pushed.

**Review by ejgallego — APPROVED** [review]
LGTM thanks. It is good to see the OCaml `trunk` build passing again.

We cannot do better easily as indicated in the discuss thread; however Dune + `utop` do support installing printers automatically by using a `ppx` annotation so it would be feasible to migrate to it at some point.

BTW I have an `utop` branch for `Drop` which is quite nice, tho it adds an extra build dep for the bytecode build.

**Comment by SkySkimmer** [comment]
@ejgallego I updated the commit message, shall we merge?

**Comment by ejgallego** [comment]
> @ejgallego I updated the commit message, shall we merge?

From my part yes, I was leaving time for the rest of the devs to comment. Taking this tentatively, will merge tomorrow / Jeudi if no more comments.
